Marlin/Marlin_main.cpp

```diff
@@ -1659,19 +1659,19 @@ static void clean_up_after_endstop_or_probe_move() {
 
       // If Z needs to raise, do it before moving XY
       if (current_position[Z_AXIS] < z) {
-        feedrate = homing_feedrate[Z_AXIS];
+        feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
         current_position[Z_AXIS] = z;
         line_to_current_position();
       }
 
-      feedrate = XY_PROBE_FEEDRATE;
+      feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
       current_position[X_AXIS] = x;
       current_position[Y_AXIS] = y;
       line_to_current_position();
 
       // If Z needs to lower, do it after moving XY
       if (current_position[Z_AXIS] > z) {
-        feedrate = homing_feedrate[Z_AXIS];
+        feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
         current_position[Z_AXIS] = z;
         line_to_current_position();
       }
```

Comment:
The delta move is 3-dimensional anyway.
The moves for the Cartesian are splittet in horizontal and vertical moves - both types will have the same feedrate - but are later limited by the axis MAX_FEEDRATE anyway. Currently the feedrate is only used for the Allen key probe moves - the users is responsible to config this correct. In case he can configure splittet moves. (In case of a i3 with M5 lead-screws you want to have MAX_FEEDRATE for z anyway - always! :-) )
Don't make it more complicated than it needs.